### ISSUE TITLE
Moving selected object automatically copies it

### ISSUE THREAD

**Issue by Septeract7** [xournalpp/xournalpp#1750]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Windows 10
 - 3.0 (i think...?)
 - Version of Xournal++: 1.0.17, commit 12febf7e
 - Installation method: official installer for windows

**Describe the bug**
If i select a bunch of handwritten text using the freehand or rectangle selection tool and then grab, move it around and let go again, the selected text stays where it is but also is copied to where i dragged the selection. 

**To Reproduce**

1. Write some text (handwritten or in a textbox)
2. Activate freehand or rectangle selection tool
3. Select text from step 1
4. Grab and drag selection to a new location
5. Let go of selection

**Expected behavior**
During step 4 you should see two versions of the selection: 1) the original selection in the original location and 2) the copied version you can drag around. 
After step 5 (letting go of the selection), version 1) should disappear and only the moved text should stay. 

**Additional context**
I am using a Wacom Intuos S tablet to draw/write on my (non-touch) laptop. All steps are performed using the pen.

**Comment by Technius** [comment]
To clarify, you're using the eraser of your pen to erase and not by manually switching to the eraser tool?

This issue might be happening because one of your pen buttons is considered to be a right click. Currently, a right click on a selection will duplicate it (so you end up dragging a duplicated item).

I can't reproduce this on my Linux installation; either I'm missing something or this is Windows-specific or setup-specific.

**Comment by LittleHuba** [comment]
This is a rendering issue. The update of the area the elements are moved from is triggered but does not happen fast enough. I assume this is caused by the document lock being held for moving the elements but I never got the time to actually check.

**Comment by Septeract7** [comment]
I am using one button of my pen that is configured in Wacom software and Xournal++ settings as eraser. No manual switching. 
**But:** I noticed that i can not always reproduce the issue so now i am uncertain if it is really caused by the eraser or something else.

**Comment by Septeract7** [comment]
Quick update:
After getting used to this i find it to be quite practical in many situations to just be able to select sth, drag it and it automatically being copied. 
Could you make this a feature?

**Comment by jvwowd** [comment]
I have noticed something similar. In my case, I have my pen configured as a mouse (I'm not sure if that matters), and one of the buttons is set to the hand tool. After selecting some objects, if I hold down the hand tool button on my pen while hovering over the selection area then hover to somewhere else the objects will automatically be copied. If not using the hand tool button and touching the pen to the surface of the tablet and dragging the selected objects, the objects are moved without being copied as would be expected.

I agree that this is convenient as a feature as the process is more streamlined than copy and paste.

**Comment by Technius** [comment]
I think the best option is to make the behavior configurable. Some users might not use the copy-paste behavior much, but for other users this is a killer feature.

**Comment by chrostino** [comment]
the issue is that for some stylus that have the right click associated with the button you use for selection it works like that (even when the selection tool is associated to that button), but for styluses that have other mouse buttons associated to the button you use for selection (like mine that has the mouse middle click) it doesn't. A workaround would be to remap that button to mouse right click, but for many non-wacom styluses it's kinda hard, so it wpuld be nice to have a feature to do that in xournal, maybe by being able to change which button does the copy thing, or just as an option for the selection tool (like clicking selected object with the selection tool automatically copies/doubles it)

**Comment by rickalex21** [comment]
I disabled everything in my mouse Tool - don't change. I can confirm it's still duplicating when I right click or left click with the selection tool enabled. I am not sure how to go about moving objects.

**Comment by jjanviers** [comment]
I have a similar problem, when opening a PDF, adding a png image to it. I can resize the image, but then when moving it, it will always leave a copy at the initial position. if I let go of the button and start dragging again, it again leaves a copy at the original start-drag position. So when moving a pic into place is impossible.

**Comment by Badhunter0303** [comment]
I have the same problem as jjanviers said above.

**Comment by HunzHurte** [comment]
Are you sure, you're not holding down any buttons on your stylus while moving the object around? Holding the select button while moving an object duplicates it in place.

**Comment by jjanviers** [comment]
> Are you sure, you're not holding down any buttons on your stylus while moving the object around? Holding the select button while moving an object duplicates it in place.

I just use mouse (logitech mx master 3 with solaar); no pens etc.

**Comment by clej37** [comment]
I got the same problem (using the touchpad of a dell laptop). Restarting the application solved the issue. Maybe there was an option locked/activated...

**Comment by TimMoser92** [comment]
> > Are you sure, you're not holding down any buttons on your stylus while moving the object around? Holding the select button while moving an object duplicates it in place.
> 
> I just use mouse (logitech mx master 3 with solaar); no pens etc.

I've got exactly the same problem with exactly the same mouse (and driver). I just installed xournal++, changed no settings, opened a pdf, added an image using the image tool and then got a copy every time I moved it.

Forgot to mention: this also happens when I use my touchpad (frame.work laptop).
